src/capabilities.js

```diff
@@ -254,6 +288,10 @@ export class Capabilities {
    * @param {keyof typeof DEFAULT_CAPABILITIES} roleId
    */
   async assignRole(deviceId, roleId) {
+    if (deviceId !== this.#ownDeviceId && roleId === LEFT_ROLE_ID) {
```

Comment:
I think it's ok to do it this way, although it does mean that we need to make it a hard requirement that all capability records can assign this (although there is the open question of whether a blocked user can also leave - they wouldn't be able to leave for other reasons, because they would not be able to write to the project any more).

An alternative approach would be to not include `LEFT_ROLE_ID` in the `roleAssignments` property and instead add this check to line 318 (`if (!ownCapabilities.roleAssignment.includes(roleId)) {`).

Reply:
addressed via https://github.com/digidem/mapeo-core-next/pull/410/commits/ca03ef35b9392a6cfd54196836ac892b30a60b89